### PR TITLE
Add RizinFunctionTask and common base class with RizinCmdTask

### PR DESCRIPTION
**Pull request by thestr4ng3r**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [ ] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**

This is the equivalent of `rz_core_function_task_new()` but for C++ with signals. Necessary for asynchronous rz-ghidra.

### CODE REVIEW

**Comment by XVilka** [comment]
You probably want to drop the gcc-5 support if you want `std::unique_ptr`:
```
In file included from ../src/widgets/GraphHorizontalAdapter.cpp:1:0:
../src/widgets/GraphHorizontalAdapter.h:13:43: error: expected ‘)’ before ‘<’ token
     GraphHorizontalAdapter(std::unique_ptr<GraphLayout> layout);
                                           ^
../src/widgets/GraphHorizontalAdapter.h:20:10: error: ‘unique_ptr’ in namespace ‘std’ does not name a template type
     std::unique_ptr<GraphLayout> layout;
```

**Comment by XVilka** [comment]
I suggest to move from 16.04 to 18.04 and add 20.04 instead.

File: src/common/RizinTask.h [hunk 1]
```diff
@@ -4,32 +4,58 @@
...
+    Q_OBJECT
+
+private:
+    std::function<void *(RzCore *)> fcn;
+    void *res;
+
+    static void *runner(RzCore *core, void *user);
+
+public:
+    explicit RizinFunctionTask(std::function<void *(RzCore *)> fcn, bool transient = true);
```
Comment (ITAYC0HEN) [inline src/common/RizinTask.h:56]: Docstrings here and other new functions, especially important ones

**Review by ITAYC0HEN — CHANGES_REQUESTED** [review]

**Comment by thestr4ng3r** [comment]
It was only missing includes.

**Pull request merged**